jax/api_util.py

```diff
@@ -74,16 +75,78 @@ def flatten_fun_nokwargs2(in_tree, *args_flat):
   aux_flat, aux_tree = tree_flatten(aux)
   yield (ans_flat, aux_flat), (ans_tree, aux_tree)
 
-def argnums_partial(f, dyn_argnums, args):
+
+def argnums_partial(f, dyn_argnums, args, raise_error=False):
+  """Keeps only dynamic arguments for the arguments of f.
+
+  Args:
+    f: The function being wrapped.
+    dync_arguments: The indices of the dynamic arguments (i.e. non-static
+      positial arguments).
+    args: The positional arguments.
+
+  Returns:
+    A tuple of:
+      The function wrapped into a partial with the non dynamic arguments
+        wrapped.
+      The tuple of the dynamic arguments.
+  """
   if isinstance(dyn_argnums, int):
     dyn_argnums = (dyn_argnums,)
-  else:
+  elif not isinstance(dyn_argnums, tuple):
     dyn_argnums = tuple(dyn_argnums)
-  fixed_args = tuple([unit if i in dyn_argnums else wrap_hashably(arg)
-                      for i, arg in enumerate(args)])
+  fixed_args = tuple(
+      unit if i in dyn_argnums else wrap_hashably(arg)
+      for i, arg in enumerate(args))
   dyn_args = tuple(args[i] for i in dyn_argnums)
   return _argnums_partial(f, dyn_argnums, fixed_args), dyn_args
 
+
+def argnums_partial_except(f,
+                           static_argnums: Tuple[int, ...],
+                           args,
+                           raise_error=False):
+  """Removes static arguments from the explicit list of arguments for f.
+
+  Args:
+    f: The function being wrapped.
+    static_argnums: The indices of the static arguments.
+    args: The positional arguments.
+    raise_error: If `True`, an error will be raised instead of a warning.
+
+  Returns:
+    A tuple of:
+      The function wrapped into a partial with static-arguments wrapped.
```

Comment:
It's not accurate to say the function is wrapped into a partial, because what's actually being returned is a lu.WrappedFun instance.

I think this docstring isn't saying much more than the type annotations above. We could either
1. adapt the language so that it's precise (like the type annotations)
2. delete most of the docstring.
I suggest the latter.

How about a one-line docstring that just says

```python
"""Version of ``argnums_partial`` that checks hashability of static_argnums."""
```

Reply:
Done.